### PR TITLE
Make rules updated/introduced

### PR DESCRIPTION
**Pull request by himanshu-kun**
**What this PR does / why we need it**:
Introduced make rule `check-vulnerabilities` and `format`

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement developer
`format` and `check-vulnerabilites` make targets introduced
```

### CODE REVIEW

**Comment by unmarshall** [comment]
Changes are fine but a lot of libs are being pulled into vendor directory. Why is that the case?

**Comment by himanshu-kun** [comment]
I have added `github.com/gardener/gardener/hack` as a required package in `hack/tools.go`. This is done so that the scripts can be vendored in.
So the dependencies listed in `github.com/gardener/gardener/hack` have also come in dwd.

File: cmd/probe.go [hunk 1]
```diff
@@ -103,7 +103,12 @@ func runProbe(cmd *cobra.Command, args []string) {
 	leaderElectionClient := kubernetes.NewForConfigOrDie(rest.AddUserAgent(config, "dependency-watchdog-election"))
 	recorder := createRecorder(leaderElectionClient)
 	run := func(ctx context.Context) {
-		go serveMetrics()
+		go func() {
+			err := serveMetrics()
+			if err != nil {
+				klog.Fatalf("Can't serve metrics: %s", err)
```
Comment (unmarshall) [inline cmd/probe.go:109]: Should this kill the container? Metrics are good to have and not must to have, right?

File: cmd/root.go [hunk 2]
```diff
@@ -157,7 +157,12 @@ func runRoot(cmd *cobra.Command, args []string) {
 	leaderElectionClient := kubernetes.NewForConfigOrDie(rest.AddUserAgent(config, "dependency-watchdog-election"))
 	recorder := createRecorder(leaderElectionClient)
 	run := func(ctx context.Context) {
-		go serveMetrics()
+		go func() {
+			err := serveMetrics()
+			if err != nil {
+				klog.Fatalf("Can't serve metrics: %s", err)
```
Comment (unmarshall) [inline cmd/root.go:163]: Similar comment as the above.

**Comment by unmarshall** [comment]
Looks good to me. Why is the unit test failing? Also build happens in concourse and test and verify build image happens in prow?

**Review by unmarshall — APPROVED** [review]
/lgtm

**Comment by unmarshall** [comment]
/do-not-merge Have a look at why unit tests are failing

File: cmd/probe.go (hunk 1 above)
Comment (himanshu-kun) [reply cmd/probe.go:109]: okay will error on this

**Comment by himanshu-kun** [comment]
PR updating `.ci/pipeline_definitions` ->  https://github.com/gardener/dependency-watchdog/pull/66

**Review by unmarshall — APPROVED** [review]

**Pull request merged**